### ISSUE TITLE
Try s3-helper sidecar with nginx in Turnkey

### ISSUE THREAD

**Issue by phuongdh** [avalonmediasystem/avalon-terraform#6]
### Description
Currently nginx uses IP-based policy to access S3. Would be better to use role-based policy with signed S3 request. s3-helper (crunchyroll.com) can do this transparently as an nginx upstream endpoint. Only caveat is it requires 127.0.0.0 origin so if we want it on a separate container, it needs forking.

https://medium.com/ellation-tech/just-in-time-vod-packaging-in-an-aws-environment-with-role-based-credentials-4e33202a4450

### Done Looks Like
- Try out the role-based policy with signed S3 request per description above and create a new issue to implement in Turnkey if it works out

**Comment by phuongdh** [comment]
Added a build stage to nginx Dockerfile but can't build because of https://github.com/crunchyroll/evs-s3helper/issues/23

```
FROM        golang:alpine as s3-helper
RUN         apk --no-cache add git
RUN         git clone --depth=1 https://github.com/crunchyroll/evs-s3helper.git
RUN         cd evs-s3helper/s3-helper \
         && go get \
         && go build
```

**Comment by phuongdh** [comment]
Managed to build and run it but getting timeouts :(

```
{"level":"info","time":1586965062,"message":"Starting up"}
{"level":"info","time":1586965062,"message":"S3Region: us-east-1"}
{"level":"info","time":1586965062,"message":"S3Bucket: avalon-demo-derivatives"}
{"level":"info","time":1586965062,"message":"LogLevel: debug"}
{"level":"info","time":1586965062,"message":"System has 4 CPUs"}
{"level":"info","time":1586965062,"message":"Setting thread concurrency to 4"}
{"level":"info","time":1586965062,"message":"Accepting connections on 0.0.0.0:8080"}
{"level":"info","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","url":"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","time":1586965066,"message":"Received request"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection timeout: retry #0"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection timeout: retry #1"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection timeout: retry #2"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection timeout: retry #3"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection timeout: retry #4"}
{"level":"error","object":"/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4","range":"","method":"GET","error":"Get \"http://s3-us-east-1.amazonaws.com/avalon-demo-derivatives/0016f5ec-d477-4d7e-9fba-a57db48c9f5b/quality-low/electricbeat-1.mp4\": dial tcp: i/o timeout","time":1586965066,"message":"Connection failed after #5 retries"}
```

**Comment by phuongdh** [comment]
I made a buildable and working fork here https://github.com/avalonmediasystem/evs-s3helper
